File: bindings/java-jni/src/main/java/org/maplibre/nativejni/error/InvalidArgumentException.java
package org.maplibre.nativejni.error;

/** Thrown when a native call reports an invalid argument. */
public final class InvalidArgumentException extends MaplibreException {
  public InvalidArgumentException(int nativeStatusCode, String diagnostic) {
    super(MaplibreStatus.INVALID_ARGUMENT, nativeStatusCode, diagnostic);
  }
}

File: bindings/dotnet/src/Maplibre.Native/Error/InvalidArgumentException.cs
namespace Maplibre.Native.Error;

/// <summary>Exception for invalid native or binding arguments.</summary>
public sealed class InvalidArgumentException : MaplibreException
{
    public InvalidArgumentException(
        MaplibreStatus status,
        int? rawStatus,
        string diagnostic,
        Exception? innerException
    )
        : base(status, rawStatus, diagnostic, innerException) { }
}
